ui/intents/src/main/java/de/danoeh/antennapod/ui/intents/AppLauncher.java

```diff
@@ -0,0 +1,80 @@
+package de.danoeh.antennapod.ui.intents;
+
+import android.app.PendingIntent;
+import android.content.Context;
+import android.content.Intent;
+import android.os.Build;
+
+/**
+ * Launches the main activities of the app with specific arguments.
+ */
+public abstract class AppLauncher {
+    public static final String INTENT_MAIN_ACTIVITY = "de.danoeh.antennapod.intents.MAIN_ACTIVITY";
+    public static final String INTENT_VIDEO_PLAYER = "de.danoeh.antennapod.intents.VIDEO_PLAYER";
+
+    public static final String EXTRA_OPEN_PLAYER = "open_player";
+
+    public static MainActivityLaunchBuilder mainActivity(Context context) {
+        return new MainActivityLaunchBuilder(context);
+    }
+
+    public static class MainActivityLaunchBuilder {
```

Comment:
The class `AppLauncher` is designed to contain almost unrelated activity builders. Wouldn't it be cleaner to keep the inner classes as separate files? The convenience methods like `mainActivity()` would then be replaced by a constructor call.

Reply:
I think a static `AppLauncher.mainActivity()` looks cooler than a constructor call but you are right that it should be separate files without the unnecessary nesting. The class will probably also grow some more (eg when adding methods for opening specific fragments).